tpm2/structures.go

```diff
@@ -256,23 +256,25 @@ type ECCParams struct {
 	Point     ECPoint
 }
 
-// ECPoint represents a ECC coordinates for a point.
+// ECPoint represents a ECC coordinates for a point using byte buffers.
 type ECPoint struct {
-	X, Y *big.Int
+	XRaw, YRaw tpmutil.U16Bytes
 }
 
-func (p *ECPoint) x() *big.Int {
-	if p == nil || p.X == nil {
-		return big.NewInt(0)
+func (p ECPoint) X() *big.Int {
+	x := big.NewInt(0)
+	if p.XRaw == nil {
+		return x
 	}
-	return p.X
+	return x.SetBytes(p.XRaw)
 }
 
-func (p *ECPoint) y() *big.Int {
-	if p == nil || p.Y == nil {
-		return big.NewInt(0)
+func (p ECPoint) Y() *big.Int {
+	y := big.NewInt(0)
```

Comment:
you can just use `new(big.Int).SetBytes(x)` and `new(big.Int).SetBytes(y)` here. The nil slice of length zero works fine with `SetBytes`.

Reply:
Done